kernel/src/managers/dma/dma-dt.c.in

```diff
@@ -86,7 +86,7 @@ static const dma_meta_t streams[] = {
             .source = {{ stream_get_transfer_source(node) }},
             .dest = {{ stream_get_transfer_dest(node) }},
             {% if node["length"] -%}
-            .transfer_len = {{ "%08u"|format(node["length"]) }}UL,
+            .transfer_len = {{ "%u"|format(node["length"]) }}UL,
             {% else -%}
             .transfer_len = 0UL,
```

Comment:
Should we really accept the absence of `streamid` or `length` nodes and silently convert their values to 0 ?

Reply:
yep. streamid is useful for mem2dev or dev2mem as it defines the HW stream config id that matches the corresponding dev flow. This is useless for mem2mem (detected by the fact that both src and dest are shm). in that last case, streamid as no meaning